DataFormats/Provenance/src/ProductRegistry.cc

```diff
@@ -558,6 +604,28 @@ namespace edm {
     return itFind->second;
   }
 
+  std::optional<std::string> ProductRegistry::aliasToModule(KindOfType kindOfType,
+                                                            TypeID const& type,
+                                                            std::string_view moduleLabel,
+                                                            std::string_view productInstanceName) const {
+    auto aliasFields = [](auto const& item) {
+      return std::tie(std::get<Transients::kKind>(item),
+                      std::get<Transients::kType>(item),
+                      std::get<Transients::kModuleLabel>(item),
+                      std::get<Transients::kProductInstanceName>(item));
+    };
+    auto const target = std::tuple(kindOfType, type, moduleLabel, productInstanceName);
+    auto found =
+        std::lower_bound(transient_.aliasToOriginal_.begin(),
+                         transient_.aliasToOriginal_.end(),
+                         target,
+                         [aliasFields](auto const& item, auto const& target) { return aliasFields(item) < target; });
+    if (found == transient_.aliasToOriginal_.end() or aliasFields(*found) != target) {
+      return std::nullopt;
+    }
+    return std::get<Transients::kAliasForModuleLabel>(*found);
```

Comment:
@Dr15Jones @wddgit 

This approach does not address with the ambiguity if an EDAlias provides e.g. `std::vector<Base>` and `std::vector<Derived>` for the same module label and product instance name, and a consumer consumes `edm::View<Base>`. I was planning to rely on the ambiguity check done also for regular products, but apparently that is done only at the time of `Event::get()` (`consumes()` alone is not sufficient).

The information on `PathsAndConsumesOfModules` will be used for module deletion (#29553; in addition to dependency graph, tracer, etc), I'm wondering if this code should actually report a dependence for each ambiguous case?

Alternatively I could think of changing already `consumes()` to trigger the ambiguity exception (but that may have other consequences).

Reply:
In a quick chat with @Dr15Jones we concluded that it could be better to move the ambiguity exception to be thrown at "the `consumes()` time". I'm looking into that next (will be a separate PR).

Reply:
The ambiguity exception can not generally be thrown at the beginJob based on consumes information, because the consumption of a product without an explicit process name can only be resolved at the `Event::get()` time.

Given that the ambiguity gets resolved based on products and indices (that are not touched by #29553) leaving this code as it is now could technically be safe for module deletion. However, I think it would be clearest for `PathsAndConsumesOfModules` to report dependencies on all producers of an ambiguous `get()` with `edm::View` of `EDAlias` products.

Reply:
> However, I think it would be clearest for `PathsAndConsumesOfModules` to report dependencies on all producers of an ambiguous `get()` with `edm::View` of `EDAlias` products.

The just-added commit adds functionality.